front/src/modules/pipeline-processes/components/BoardActionBarButtonDeletePipelineProcesses.tsx

```diff
@@ -0,0 +1,47 @@
+import { getOperationName } from '@apollo/client/utilities';
+import { useRecoilState } from 'recoil';
+
+import { EntityTableActionBarButton } from '@/ui/components/table/action-bar/EntityTableActionBarButton';
+import { IconTrash } from '@/ui/icons/index';
+import { useDeleteManyPipelineProgressMutation } from '~/generated/graphql';
+
+import { GET_PIPELINES } from '../queries';
+import { boardItemsState } from '../states/boardItemsState';
+import { selectedBoardItemsState } from '../states/selectedBoardItemsState';
+
+export function BoardActionBarButtonDeletePipelineProcesses() {
+  const [selectedBoardItems, setSelectedBoardItems] = useRecoilState(
+    selectedBoardItemsState,
+  );
+  const [items, setItems] = useRecoilState(boardItemsState);
+
+  const [deletePipelineProcesses] = useDeleteManyPipelineProgressMutation({
+    refetchQueries: [getOperationName(GET_PIPELINES) ?? ''],
+  });
+
+  async function handleDeleteClick() {
+    await deletePipelineProcesses({
+      variables: {
+        ids: selectedBoardItems,
+      },
+    });
+
+    setItems(
```

Comment:
This seems unusual to manipulate dictionaries, is it possible to use an array structure ?

Reply:
`boardItems` are accessed by ID, for example to fill the columns:

https://github.com/twentyhq/twenty/blob/3b82ed913827dd208da44cbd7ad7c75f66a98065/front/src/modules/pipeline-progress/components/Board.tsx#L147

I suppose this is the reason we indexed them by ID.

To make sure I understand the comment: what is unusual ? The fact they are stored as a dict in the state or the method used to filter the dict by keys here ?